composer.json

```diff
@@ -18,7 +18,8 @@
     },
     "require-dev": {
         "phpunit/phpunit": "^8.0|^9.3",
-        "php-http/client-integration-tests": "^3.0"
+        "php-http/client-integration-tests": "^3.0",
+        "phpspec/prophecy-phpunit": "^2.0"
```

Comment:
can you allow version 1 of this as well? or does that not provide the trait? the php 7.2 tests can't install this.

if there is no easy solution, i think we should just drop the 7.2 build and instead do the lowest version build with 7.3 (and adjust composer.json accordingly)

Reply:
Oh, just tried with version 1 of the package, but sadly the trait isn't available there.
Hm, wouldn't upgrading the minimum PHP version require a major version of the package?

As prophesize is only deprecated in phpunit 9, I'd suggest the next major version of the package should require a higher PHP version (maybe 7.4 or even 8?) and upgrade phpunit. And I'd suggest that I for now revert this PR to only introduce the `composer test` command?

Reply:
dropping old PHP versions is not a BC break. the new minor version simply won't get pulled in during the composer upgrade.

but i am ok with reverting that change as its just a deprecation and not an error.

Reply:
Ah 🤔 hm, that's true and as 7.2 and 7.3 are already end of life, one could argue that it's OK that users of these PHP versions won't even receive bugfixes anymore. In any other case I'd consider it a breaking change, as users can't get any updates of the package anymore without upgrading PHP.
As you're OK with the bump to 7.3 I'll do that 👍🏻

Reply:
thanks.

dependency bumps and semver seems a discussion that is not everyone agrees how to handle. to my understanding, the main premise of semver is that if you do `composer update` without changing the constraints, you still have a working system without having to change things in your application. a minimum PHP version change does not violate this, as composer detects it. indeed you won't get bugfixes anymore (unless we would backport them to the old minor version, which is something i occasionally did for important security fixes), but that would be the same if we do a new major version.